src/main/java/io/pinecone/clients/AsyncIndex.java

```diff
@@ -846,6 +849,51 @@ public ListenableFuture<DescribeIndexStatsResponse> describeIndexStats(Struct fi
         return asyncStub.describeIndexStats(describeIndexStatsRequest);
     }
 
+    @Override
+    public ListenableFuture<ListResponse>  list(String namespace) {
```

Comment:
We also need docstrings in this file, you should be able to lift your docstrings from `Index` and put them here and update `Index` -> `AsyncIndex`.

Appreciate you adding the variety of overrides here. We discussed offline but we should really look at implementing some builder patterns for some of these more complicated interfaces at some point.

Reply:
Oops yes, thank you!

(And as discussed offline, I'll make a separate ticket to follow up this PR with a transform of these methods into a Builder pattern)